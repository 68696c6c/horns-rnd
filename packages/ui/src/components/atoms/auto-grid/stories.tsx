/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Breakpoint } from '../../../config'
import { AutoGrid, AutoGridProps } from '.'

export default {
  title: 'Atoms/AutoGrid',
  component: AutoGrid,
} as Meta

const Template: Story<AutoGridProps> = ({
  breakpoint,
  children,
  ...others
}: AutoGridProps) => (
  <>
    <p>
      Above the {breakpoint} breakpoint, these items will rendered in an
      auto-sized, wrapping grid.
    </p>
    <AutoGrid breakpoint={breakpoint} {...others}>
      {children}
    </AutoGrid>
  </>
)

export const Default = Template.bind({})
Default.args = {
  breakpoint: Breakpoint.Mobile,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}

export const Gapped = Template.bind({})
Gapped.args = {
  breakpoint: Breakpoint.Mobile,
  gapped: true,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}

export const SingleItem = Template.bind({})
SingleItem.args = {
  breakpoint: Breakpoint.Mobile,
  children: <div>one</div>,
}
