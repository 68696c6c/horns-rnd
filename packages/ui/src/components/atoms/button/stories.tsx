import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'

import { clickableButtonDemo } from '../../../_story'
import { Colorway, Font, Size } from '../../../config'
import { ButtonProps } from '../../quarks'

import { Button } from '.'

export default {
  title: 'Atoms/Button',
  component: Button,
} as Meta

const Template: Story<ButtonProps> = ({ children, ...others }: ButtonProps) => (
  <Button {...others}>{children}</Button>
)

export const Default = Template.bind({})
Default.args = {
  children: 'Button',
}

export const Props = Template.bind({})
Props.args = {
  ...Default.args,
  border: { all: { width: Size.Small } },
  color: Colorway.Primary,
  padding: { all: Size.Small },
  radius: { all: Size.Large },
  font: Font.Emphasized,
}

export const Traits = () => clickableButtonDemo(Default, { ...Default.args })
