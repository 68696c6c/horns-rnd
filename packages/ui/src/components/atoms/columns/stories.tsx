/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Breakpoint } from '@horns/theme'

import { StyledCode } from '../../../_story'

import { Columns, ColumnsProps } from '.'

export default {
  title: 'Atoms/Columns',
  component: Columns,
} as Meta

const Template: Story<ColumnsProps> = ({
  breakpoint,
  columns,
  children,
  ...others
}: ColumnsProps) => (
  <>
    <p>
      Above the {breakpoint} breakpoint, these items will be rendered as
      columns. By default, a single row of columns will be rendered with each
      item in its own column. The <StyledCode>columns</StyledCode> prop will
      limit the number of columns per row and create additional rows to fit the
      remaining items.
    </p>
    <Columns breakpoint={breakpoint} columns={columns} {...others}>
      {children}
    </Columns>
  </>
)

export const Default = Template.bind({})
Default.args = {
  breakpoint: Breakpoint.Mobile,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}

export const Gapped = Template.bind({})
Gapped.args = {
  breakpoint: Breakpoint.Mobile,
  gapped: true,
  children: (
    <>
      <div style={{ background: 'tan' }}>one</div>
      <div style={{ background: 'tan' }}>two</div>
      <div style={{ background: 'tan' }}>three</div>
      <div style={{ background: 'tan' }}>four</div>
      <div style={{ background: 'tan' }}>five</div>
      <div style={{ background: 'tan' }}>six</div>
    </>
  ),
}

export const SingleItem = Template.bind({})
SingleItem.args = {
  breakpoint: Breakpoint.Mobile,
  children: <div>one</div>,
}

export const FixedColumns = Template.bind({})
FixedColumns.args = {
  breakpoint: Breakpoint.Mobile,
  columns: 4,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}
