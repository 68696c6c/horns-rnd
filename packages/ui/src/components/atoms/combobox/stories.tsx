/* eslint-disable react/jsx-one-expression-per-line,jsx-a11y/label-has-associated-control */
import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Combobox, ComboboxProps } from '.'

export default {
  title: 'Atoms/Combobox',
  component: Combobox,
} as Meta

const options = [
  { label: 'One', value: '1' },
  { label: 'Two', value: '2' },
  { label: 'Three', value: '3' },
  { label: 'Four', value: '4' },
  { label: 'Five', value: '5' },
]

const Template: Story<ComboboxProps> = (props: ComboboxProps) => (
  <Combobox {...props} />
)

export const Default = Template.bind({})
Default.args = {
  id: 'example-select',
  name: 'example_select',
  placeholder: 'Select one...',
  options,
}

export const Multiselect = Template.bind({})
Multiselect.args = {
  ...Default.args,
  id: 'example-multiselect',
  name: 'example_multiselect',
  placeholder: 'Select one or more...',
  multiple: true,
}
