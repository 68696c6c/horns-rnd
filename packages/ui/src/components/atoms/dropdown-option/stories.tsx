import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { DropdownOption } from '.'

export default {
  title: 'Atoms/DropdownOption',
  component: DropdownOption,
} as Meta

export const Default: Story = () => <DropdownOption>hello world</DropdownOption>
