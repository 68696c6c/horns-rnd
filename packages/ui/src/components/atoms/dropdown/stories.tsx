import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Dropdown } from '.'

export default {
  title: 'Atoms/Dropdown',
  component: Dropdown,
} as Meta

export const Default: Story = () => (
  <Dropdown open>
    <li>item one</li>
    <li>item two</li>
    <li>item three</li>
  </Dropdown>
)
