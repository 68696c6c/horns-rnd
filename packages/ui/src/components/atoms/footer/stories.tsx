import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Colorway, Size } from '@horns/theme'

import { blockDemo } from '../../../_story'
import { BlockProps } from '../../quarks'

import { Footer } from '.'

export default {
  title: 'Atoms/Footer',
  component: Footer,
} as Meta

const Template: Story<BlockProps> = ({ children, ...others }: BlockProps) => (
  <Footer {...others}>{children}</Footer>
)

export const Default = Template.bind({})
Default.args = {
  children: 'Footer',
}

export const Props = Template.bind({})
Props.args = {
  ...Default.args,
  color: Colorway.Primary,
  padding: { all: Size.Small },
  fluid: false,
}

export const Traits = () => blockDemo(Footer, {})
