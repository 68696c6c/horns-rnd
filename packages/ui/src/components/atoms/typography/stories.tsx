import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'

import {
  chromaticTextDemo,
  StyledDemo,
  StyledDivider,
  StyledPre,
  typographicDemo,
} from '../../../_story'
import { Colorway, Font, HeadingLevel } from '../../../config'

import {
  Heading as HeadingAtom,
  SubHeading as SubHeadingAtom,
  T as TypographyAtom,
  TypographyProps,
} from '.'

export default {
  title: 'Atoms/Typography',
  component: TypographyAtom,
} as Meta

const Template: Story<TypographyProps> = ({
  children,
  ...others
}: TypographyProps) => <TypographyAtom {...others}>{children}</TypographyAtom>

export const T = Template.bind({})
T.args = {
  children: 'Typography',
}

export const Props = Template.bind({})
Props.args = {
  ...T.args,
  color: Colorway.Primary,
  font: Font.Emphasized,
}

export const Traits: Story = () => (
  <>
    {chromaticTextDemo(TypographyAtom, {})}
    {typographicDemo(TypographyAtom, {})}
  </>
)

export const Fonts: Story = () => (
  <>
    <h1>Fonts</h1>
    <StyledPre>{`<T font="h1">example</T>`}</StyledPre>
    <StyledDivider />
    <StyledDemo>
      {Object.values(HeadingLevel).map((variant) => (
        <TypographyAtom font={variant} key={`typography-${variant}`}>
          {variant}
        </TypographyAtom>
      ))}
      {Object.values(Font).map((variant) =>
        variant === Font.Heading ? (
          <></>
        ) : (
          <TypographyAtom font={variant} key={`typography-${variant}`}>
            {variant}
          </TypographyAtom>
        ),
      )}
    </StyledDemo>
  </>
)

export const Heading: Story<TypographyProps> = ({
  children,
  ...others
}: TypographyProps) => <HeadingAtom {...others}>{children}</HeadingAtom>
Heading.args = {
  children: 'Heading',
}

export const SubHeading: Story<TypographyProps> = ({
  children,
  ...others
}: TypographyProps) => <SubHeadingAtom {...others}>{children}</SubHeadingAtom>
SubHeading.args = {
  children: 'SubHeading',
}
