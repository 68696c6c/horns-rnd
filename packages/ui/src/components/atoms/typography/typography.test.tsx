import React from 'react'

import { render } from '../../test'
import { Colorway, Font, HeadingLevel } from '../../../config'

import { T, Props, Heading, SubHeading } from './stories'

describe('T', () => {
  it('should render as default', () => {
    const { container } = render(<T>example</T>)
    expect(container).toMatchSnapshot()
  })
  it('should support trait props', () => {
    const { container } = render(<Props {...Props.args}>example</Props>)
    expect(container).toMatchSnapshot()
  })
  it.each([...Object.values(Font), ...Object.values(HeadingLevel)])(
    'should render variant %s',
    (font) => {
      const { container } = render(<T font={font}>{font}</T>)
      expect(container).toMatchSnapshot()
    },
  )
})

describe('Heading', () => {
  it('should render as default', () => {
    const { container } = render(<Heading>example</Heading>)
    expect(container).toMatchSnapshot()
  })
  it('should support trait props', () => {
    const { container } = render(
      <Heading color={Colorway.Primary}>example</Heading>,
    )
    expect(container).toMatchSnapshot()
  })
})

describe('SubHeading', () => {
  it('should render as default', () => {
    const { container } = render(<SubHeading>example</SubHeading>)
    expect(container).toMatchSnapshot()
  })
  it('should support trait props', () => {
    const { container } = render(
      <SubHeading color={Colorway.Primary}>example</SubHeading>,
    )
    expect(container).toMatchSnapshot()
  })
})
