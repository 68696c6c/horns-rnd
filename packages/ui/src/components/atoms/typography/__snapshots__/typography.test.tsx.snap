// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render as default 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 3.21em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h1
    class="emotion-0"
  >
    example
  </h1>
</div>
`;

exports[`Heading should support trait props 1`] = `
.emotion-0 {
  color: rgb(255, 170, 0);
  text-decoration-color: rgb(255, 170, 0);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 3.21em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h1
    class="emotion-0"
    color="primary"
  >
    example
  </h1>
</div>
`;

exports[`SubHeading should render as default 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 1.64em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h3
    class="emotion-0"
  >
    example
  </h3>
</div>
`;

exports[`SubHeading should support trait props 1`] = `
.emotion-0 {
  color: rgb(255, 170, 0);
  text-decoration-color: rgb(255, 170, 0);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 1.64em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h3
    class="emotion-0"
    color="primary"
  >
    example
  </h3>
</div>
`;

exports[`T should render as default 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    example
  </span>
</div>
`;

exports[`T should render variant blockquote 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  border-left: 0.25em solid;
  margin: 1em;
  padding-left: 1em;
}

<div>
  <blockquote
    class="emotion-0"
  >
    blockquote
  </blockquote>
</div>
`;

exports[`T should render variant button 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    button
  </span>
</div>
`;

exports[`T should render variant caption 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 200;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    caption
  </span>
</div>
`;

exports[`T should render variant code 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Monaco,monospace;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <code
    class="emotion-0"
  >
    code
  </code>
</div>
`;

exports[`T should render variant control 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    control
  </span>
</div>
`;

exports[`T should render variant del 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: line-through;
  text-decoration-style: solid;
  text-transform: none;
}

<div>
  <del
    class="emotion-0"
  >
    del
  </del>
</div>
`;

exports[`T should render variant emphasized 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <em
    class="emotion-0"
  >
    emphasized
  </em>
</div>
`;

exports[`T should render variant h1 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 3.21em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h1
    class="emotion-0"
  >
    h1
  </h1>
</div>
`;

exports[`T should render variant h2 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 2.36em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h2
    class="emotion-0"
  >
    h2
  </h2>
</div>
`;

exports[`T should render variant h3 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 1.64em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h3
    class="emotion-0"
  >
    h3
  </h3>
</div>
`;

exports[`T should render variant h4 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 1.29em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h4
    class="emotion-0"
  >
    h4
  </h4>
</div>
`;

exports[`T should render variant h5 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 1.15em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h5
    class="emotion-0"
  >
    h5
  </h5>
</div>
`;

exports[`T should render variant h6 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: 1em;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <h6
    class="emotion-0"
  >
    h6
  </h6>
</div>
`;

exports[`T should render variant heading 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <span
    class="emotion-0"
  >
    heading
  </span>
</div>
`;

exports[`T should render variant kbd 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Monaco,monospace;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <kbd
    class="emotion-0"
  >
    kbd
  </kbd>
</div>
`;

exports[`T should render variant label 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    label
  </span>
</div>
`;

exports[`T should render variant legal 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Times New Roman,serif;
  font-kerning: normal;
  font-size: 0.75em;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <small
    class="emotion-0"
  >
    legal
  </small>
</div>
`;

exports[`T should render variant link 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: solid;
  text-transform: none;
  cursor: pointer;
}

.emotion-0:disabled,
.emotion-0.inactive {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: solid;
  text-transform: none;
}

.emotion-0:not(:disabled):not(.inactive):hover,
.emotion-0:not(:disabled):not(.inactive).hover {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: solid;
  text-transform: none;
}

.emotion-0:not(:disabled):not(.inactive):active,
.emotion-0:not(:disabled):not(.inactive).active {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: double;
  text-transform: none;
}

.emotion-0:not(:disabled):not(.inactive):focus,
.emotion-0:not(:disabled):not(.inactive).focus {
  outline-style: dotted;
  outline-color: inherit;
  outline-width: 1px;
  outline-offset: 1px;
}

.emotion-0:visited {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: solid;
  text-transform: none;
}

<div>
  <a
    class="emotion-0"
  >
    link
  </a>
</div>
`;

exports[`T should render variant message 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    message
  </span>
</div>
`;

exports[`T should render variant mistake 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: dotted;
  text-transform: none;
}

<div>
  <u
    class="emotion-0"
  >
    mistake
  </u>
</div>
`;

exports[`T should render variant nav 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    nav
  </span>
</div>
`;

exports[`T should render variant paragraph 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: justify;
  text-decoration-line: none;
  text-transform: none;
  margin: 0 0 1em 0;
}

<div>
  <p
    class="emotion-0"
  >
    paragraph
  </p>
</div>
`;

exports[`T should render variant pre 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Monaco,monospace;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <pre
    class="emotion-0"
  >
    pre
  </pre>
</div>
`;

exports[`T should render variant quote 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <q
    class="emotion-0"
  >
    quote
  </q>
</div>
`;

exports[`T should render variant s 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <s
    class="emotion-0"
  >
    s
  </s>
</div>
`;

exports[`T should render variant samp 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <samp
    class="emotion-0"
  >
    samp
  </samp>
</div>
`;

exports[`T should render variant strong 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 200;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <strong
    class="emotion-0"
  >
    strong
  </strong>
</div>
`;

exports[`T should render variant sub 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <sub
    class="emotion-0"
  >
    sub
  </sub>
</div>
`;

exports[`T should render variant sup 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <sup
    class="emotion-0"
  >
    sup
  </sup>
</div>
`;

exports[`T should render variant table 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    table
  </span>
</div>
`;

exports[`T should render variant text 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <span
    class="emotion-0"
  >
    text
  </span>
</div>
`;

exports[`T should render variant u 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: underline;
  text-decoration-style: dotted;
  text-transform: none;
}

<div>
  <u
    class="emotion-0"
  >
    u
  </u>
</div>
`;

exports[`T should render variant variable 1`] = `
.emotion-0 {
  color: rgb(1, 1, 1);
  text-decoration-color: rgb(1, 1, 1);
  font-family: Monaco,monospace;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <var
    class="emotion-0"
  >
    variable
  </var>
</div>
`;

exports[`T should support trait props 1`] = `
.emotion-0 {
  color: rgb(255, 170, 0);
  text-decoration-color: rgb(255, 170, 0);
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

<div>
  <em
    class="emotion-0"
    color="primary"
  >
    example
  </em>
</div>
`;
